pkg/session/session.go

```diff
@@ -197,11 +206,11 @@ func GetOrCreate(ctx context.Context, params *Params) (*Session, error) {
 	return &session, nil
 }
 
-func newClient(ctx context.Context, logger logr.Logger, sessionKey string, url *url.URL, thumbprint string, feature Feature) (*govmomi.Client, error) {
-	insecure := thumbprint == ""
+func newClient(ctx context.Context, logger logr.Logger, sessionKey string, url *url.URL, params *Params) (*govmomi.Client, error) {
```

Comment:
wondering if we should do this function signature changes 🤔 

its more transparent in whats getting used from the input with the previous version. (also for `newClient`)

Reply:
As far as I can tell we need every single field in params. So this seems simplre than passing in everything individually

Reply:
Although I would now prefer doing this: https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/2154#discussion_r1284289759 and then reverting the changes to newManager & newClient

Reply:
I totally missed out that they were all used because I did not take the logger things to consideration 😂 👍